File: modules/error-tracking/impl/sentry-go/sentry.go
// Package sentry implements the ErrorTrackingService interface using Sentry.
package sentry

import (
	"context"

	contracts "github.com/dat2503/modkit/contracts/go"
)

// Config holds the configuration for the Sentry error tracking provider.
type Config struct {
	// DSN is the Sentry Data Source Name from your project settings.
	DSN string

	// Environment is the environment tag (e.g. "production", "staging").
	Environment string

	// TracesSampleRate is the fraction of transactions to capture (0.0–1.0).
	TracesSampleRate float64
}

// Service implements contracts.ErrorTrackingService using Sentry.
type Service struct {
	cfg Config
	// TODO: add sentry-go hub
}

// New creates a new Sentry error tracking service.
func New(cfg Config) *Service {
	return &Service{cfg: cfg}
}

func (s *Service) CaptureError(ctx context.Context, err error, opts contracts.CaptureOptions) error {
	// TODO: implement using github.com/getsentry/sentry-go sentry.CaptureException(err)
	panic("not implemented")
}

func (s *Service) CaptureMessage(ctx context.Context, msg string, level contracts.ErrorLevel, opts contracts.CaptureOptions) error {
	// TODO: implement using sentry.CaptureMessage(msg) with level
	panic("not implemented")
}

func (s *Service) SetUser(ctx context.Context, user contracts.ErrorUser) context.Context {
	// TODO: implement using sentry.ConfigureScope(func(scope) { scope.SetUser(...) })
	panic("not implemented")
}

func (s *Service) Flush(ctx context.Context) error {
	// TODO: implement using sentry.Flush(timeout)
	panic("not implemented")
}

File: modules/error-tracking/impl/sentry-ts/sentry.ts
import type { IErrorTrackingService, CaptureOptions, ErrorUser, ErrorLevel } from '../../../contracts/ts/errors'

export interface SentryConfig {
  dsn: string
  environment?: string
  tracesSampleRate?: number
}

/**
 * SentryErrorTrackingService implements IErrorTrackingService using Sentry.
 */
export class SentryErrorTrackingService implements IErrorTrackingService {
  constructor(private readonly config: SentryConfig) {}

  async captureError(err: Error, opts?: CaptureOptions): Promise<void> {
    // TODO: implement using @sentry/node Sentry.captureException(err)
    console.warn('[sentry] stub: captureError() not implemented')
  }

  async captureMessage(msg: string, level: ErrorLevel, opts?: CaptureOptions): Promise<void> {
    // TODO: implement using @sentry/node Sentry.captureMessage(msg, level)
    console.warn('[sentry] stub: captureMessage() not implemented')
  }

  setUser(user: ErrorUser): void {
    // TODO: implement using Sentry.setUser({ id, email })
    console.warn('[sentry] stub: setUser() not implemented')
  }

  clearUser(): void {
    // TODO: implement using Sentry.setUser(null)
    console.warn('[sentry] stub: clearUser() not implemented')
  }

  async flush(timeoutMs = 2000): Promise<void> {
    // TODO: implement using Sentry.flush(timeoutMs)
    console.warn('[sentry] stub: flush() not implemented')
  }
}
